File: tiny-event-sourcing-spring-app/src/main/kotlin/ru/quipy/shop/order/events/OrderProductAddedEvent.java
package ru.quipy.shop.order.events;


import lombok.Getter;
import ru.quipy.core.annotations.DomainEvent;
import ru.quipy.domain.Event;
import ru.quipy.shop.order.entities.Product;
import ru.quipy.shop.order.OrderAggregate;

import java.util.UUID;


@DomainEvent(name = "PRODUCT_ADDED_EVENT")
public class OrderProductAddedEvent extends Event<OrderAggregate> {

    @Getter
    private final UUID orderId;

    @Getter
    private final Product product;

    public OrderProductAddedEvent(UUID orderId, Product product) {
        super();
        this.setCreatedAt(System.currentTimeMillis());
        this.orderId = orderId;
        this.product = product;
    }
}

File: tiny-event-sourcing-spring-app/src/main/kotlin/ru/quipy/shop/order/events/OrderProductAddedEvent.kt
package ru.quipy.shop.order.events


import lombok.Getter
import ru.quipy.core.annotations.DomainEvent
import ru.quipy.domain.Event
import ru.quipy.shop.order.entities.Product
import ru.quipy.shop.order.OrderAggregate

import java.util.UUID


@DomainEvent(name = "PRODUCT_ADDED_EVENT")
class OrderProductAddedEvent(
    @Getter
    private val orderId: UUID,
    @Getter val productId: UUID,
    val quantity: Int,
    val price: Long
) : Event<OrderAggregate>(
    name = "PRODUCT_ADDED_EVENT"
)